R/class-workbook.R

```diff
@@ -5400,6 +5728,281 @@ wbWorkbook <- R6::R6Class(
       max(private$get_sheet_id(type = "sheetId", i = i), 0L, na.rm = TRUE) + 1L
     },
 
+    do_conditional_formatting = function(
+        sheet,
+        startRow,
+        endRow,
+        startCol,
+        endCol,
+        dxfId,
+        formula,
+        type,
+        values,
+        params
+    ) {
+      # TODO consider defaults for logicals
+      # TODO rename: setConditionFormatting?  Or addConditionalFormatting
+      # TODO can this be moved to the sheet data?
+      sheet <- private$get_sheet_index(sheet)
+      sqref <- stri_join(
+        get_cell_refs(data.frame(x = c(startRow, endRow), y = c(startCol, endCol))),
+        collapse = ":"
+      )
+
+      ## Increment priority of conditional formatting rule
+      for (i in rev(seq_along(self$worksheets[[sheet]]$conditionalFormatting))) {
+        priority <- reg_match0(
+          self$worksheets[[sheet]]$conditionalFormatting[[i]],
+          '(?<=priority=")[0-9]+'
+        )
+        priority_new <- as.integer(priority) + 1L
+        priority_pattern <- sprintf('priority="%s"', priority)
+        priority_new <- sprintf('priority="%s"', priority_new)
+
+        ## now replace
+        self$worksheets[[sheet]]$conditionalFormatting[[i]] <- gsub(
+          priority_pattern,
+          priority_new,
+          self$worksheets[[sheet]]$conditionalFormatting[[i]],
+          fixed = TRUE
+        )
+      }
+
+      nms <- c(names(self$worksheets[[sheet]]$conditionalFormatting), sqref)
+      dxfId <- max(dxfId, 0L)
+
+      # big switch statement
+      cfRule <- switch(
+        type,
+
+        colorScale = {
+
+          ## formula contains the colours
+          ## values contains numerics or is NULL
+          ## dxfId is ignored
+
+          if (is.null(values)) {
+            # could use a switch() here for length to also check against other
+            # lengths, if these aren't checked somewhere already?
+            if (length(formula) == 2L) {
+              sprintf(
+                # TODO is this indentation necessary?
```

Comment:
Haha, was starring at it and thought the same. No, absolutely not. Can be indented in a sane way, or not at all.

Reply:
Okay, that's good to know.  I couldn't tell if the additional spacing was necessary or just from some copy-paste.  We can probably safely clean this up later.

Reply:
Sure!